electrum/wallet.py

```diff
@@ -3634,6 +3636,9 @@ def __new__(self, db: 'WalletDB', storage: Optional[WalletStorage], *, config: S
         wallet_type = db.get('wallet_type')
         WalletClass = Wallet.wallet_class(wallet_type)
         wallet = WalletClass(db, storage, config=config)
+        # if file exists, consolidate pending changes
+        if storage and storage.file_exists() and db.pending_changes:
+            db._write(storage)
```

Comment:
Shouldn't this be done in Abstract_Wallet.__init__ instead? At least in the unit tests, `Wallet()` is not always used -- e.g. we might construct a `Standard_Wallet` or a `Multisig_Wallet` directly.

Reply:
Indeed this was not the right place.
This code is now in `JsonDB.load_data`, called from `__init__`.